packages/connect-webextension/src/index.ts

```diff
@@ -113,6 +127,15 @@ const init = (settings: Partial<ConnectSettings> = {}): Promise<void> => {
 const call: CallMethod = async params => {
     logger.debug('call', params);
 
+    if (_popupManager.popupWindow?.id) {
+        const currentPopupExists = await checkIfTabExists(_popupManager.popupWindow?.id);
+        if (!currentPopupExists) {
+            // popupWindow has reference but it does not exists so we clear it.
+            _popupManager.clear();
```

Comment:
Yes, it is other approach. In connect-web we use the `unload` event that is actually deprecated according to this https://developer.mozilla.org/en-US/docs/Web/API/Window/unload_event . There is also `beforeunload` that is not deprecated and might works as well.

On the other side using the approach of checking `checkIfTabExists` guarantee us that if the tab does not exists we create new one.

Or maybe a combination of both approaches is best.

Reply:
not sure if we are on the same page here. the `unload` event does not seem related to me here. 
can't we simply have
- some class PopupManager
- this class has method "open"
- method open runs a check whether tab exists in itself?
maybe I am missing something

Reply:
I am thinking..... The reason I am doing it here instead of in PopupManager is because of `handshakePromise` is defined here because it is required to create a new `createDeferred` as you can see here in line 135.

I am now thinking if `handshakePromise` could be part of PopupManager...

Reply:
@mroz22 what about https://github.com/trezor/trezor-suite/pull/10426/commits/b96fb8c4ca0df8da1acef98cff48c5c93cc7dd85  ?